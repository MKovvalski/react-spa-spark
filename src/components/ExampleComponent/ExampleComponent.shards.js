import styled   from '@emotion/styled'
import { down } from 'styled-breakpoints'

export const PositionWrapper = styled.div`
  width: 100%;
  height: 100vh;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  padding: 10px;
  font-size: 20px;
  margin-top: 15px;
`

export const MainTitle = styled.h3`
  font-size: ${({ theme }) => theme.contentSize.header};
  color: ${({ theme }) => theme.colors.primary};
  margin-bottom: 15px;
  transition: color .5s;
  ${down('lg')} {
    color: ${({ theme }) => theme.colors.secondary};
  }
`
export const WhatsNew = styled.ul`
  list-style-type: circle;
  font-size: ${({ theme }) => theme.contentSize.list};
  margin-bottom: 15px;
  color: ${({ toggled, theme }) => toggled ? theme.colors.primary : theme.colors.defaultBlack};
`

export const ListElement = styled.li`
  margin-bottom: 5px;
  font-family: ${({ theme }) => theme.fonts.secondary};
`
