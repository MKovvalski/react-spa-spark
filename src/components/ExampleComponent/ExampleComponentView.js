import React, { useState } from 'react'
import {
  MainTitle,
  PositionWrapper,
  WhatsNew,
  ListElement
}                          from './ExampleComponent.shards'
import { CounterButton }   from '../Commons/Buttons.shards'

const ExampleComponent = () => {

  const [toggled, toggleColor] = useState(false)
  return (
    <PositionWrapper>
      <MainTitle>
        Aloha and welcome to React SPA spark!
      </MainTitle>
      <WhatsNew toggled={toggled}>
        <ListElement>emotion replaced sass</ListElement>
        <ListElement>stylelint replaced sass-lint</ListElement>
        <ListElement>removed code overhead from webpack configs</ListElement>
        <ListElement>created a example file structure for emotion</ListElement>
      </WhatsNew>
      <CounterButton onClick={() => toggleColor(!toggled)}>
        I am a common button
      </CounterButton>
    </PositionWrapper>
  )
}

export default ExampleComponent
