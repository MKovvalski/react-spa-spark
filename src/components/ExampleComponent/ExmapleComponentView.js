import React, { useState } from 'react'

const ExampleComponent = () => {

  const [count, setCount] = useState(0)

  // TODO: add state test
  // add base methods for shallow and so forth
  // test conditional rendering
  return (
    <div className='example-component-class'>
      <h2>I am an Example Component</h2>
      <div>Build to test and showcase capabilities of jest and enzyme</div>
      <div
        className='example-component-class__button'
        onClick={() => setCount(count + 1)}
      >
        Up the Counter!
      </div>
      <div>
        Display number of counter clicks:
        <span>{count}</span>
      </div>
    </div>
  )
}

export default ExampleComponent
