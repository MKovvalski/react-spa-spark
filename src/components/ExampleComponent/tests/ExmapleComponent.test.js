import React             from 'react'
import { render, mount } from 'enzyme'
import ExampleComponent  from '../index'

describe('ExampleComponent', () => {

  const component = mount(<ExampleComponent />)

  it('renders component with counter set to 0', () => {
    expect(component.find('span').text()).toEqual('0')
  })

  it('responds to click action with counter update', () => {
    component
      .find('.example-component-class__button')
      .simulate('click')

    expect(component.find('span').text()).toEqual('1')
  })

  it('component renders correctly', () => {

    const component = render(<ExampleComponent />)
    expect(component).toMatchSnapshot()
  })
})
