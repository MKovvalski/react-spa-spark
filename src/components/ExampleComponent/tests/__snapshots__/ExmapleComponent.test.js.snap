// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExampleComponent component renders correctly 1`] = `
<div
  class="example-component-class"
>
  <h2>
    I am an Example Component
  </h2>
  <div>
    Build to test and showcase capabilities of jest and enzyme
  </div>
  <div
    class="example-component-class__button"
  >
    Up the Counter!
  </div>
  <div>
    Display number of counter clicks:
    <span>
      0
    </span>
  </div>
</div>
`;
